fabric-model-loading-api-v1/src/testmodClient/java/net/fabricmc/fabric/test/model/loading/ModelTestModClient.java

```diff
@@ -25,15 +33,38 @@
 import net.fabricmc.fabric.api.client.rendering.v1.LivingEntityFeatureRendererRegistrationCallback;
 import net.fabricmc.fabric.api.resource.ResourceManagerHelper;
 
+import net.minecraft.util.math.Direction;
+import net.minecraft.util.math.random.Random;
+
+import java.util.List;
+
 public class ModelTestModClient implements ClientModInitializer {
 	public static final String ID = "fabric-model-loading-api-v1-testmod";
 
 	public static final Identifier MODEL_ID = new Identifier(ID, "half_red_sand");
 
+	static class DownQuadRemovingModel extends ForwardingBakedModel {
+		public DownQuadRemovingModel(BakedModel model) {
+			wrapped = model;
+		}
+
+		@Override
+		public List<BakedQuad> getQuads(BlockState blockState, Direction face, Random rand) {
+			return face == Direction.DOWN ? ImmutableList.of() : super.getQuads(blockState, face, rand);
+		}
+	}
+
 	@Override
 	public void onInitializeClient() {
 		ModelLoadingPlugin.register(pluginContext -> {
 			pluginContext.addModel(MODEL_ID);
+			pluginContext.onBakedModelLoad().register((model, context) -> {
+				if(context.location().getPath().equals("block/dirt")) {
+					return new DownQuadRemovingModel(model);
```

Comment:
Very nice - could you add similar tests for the other two hooks - just for completeness?

Reply:
Yes, added.